jscomp/runtime/js_string.ml

```diff
@@ -106,326 +101,363 @@ external get : t -> int -> t = ""
 ]}
 *)
 
-external charAt : int -> t = "charAt"
-[@@mel.send.pipe: t]
-(** [charAt n s] gets the character at index [n] within string [s]. If [n] is negative or greater than the length of [s], returns the empty string. If the string contains characters outside the range [\u0000-\uffff], it will return the first 16-bit value at that position in the string.
+external charAt : t -> pos:int -> t = "charAt"
+[@@mel.send]
+(** [charAt s ~pos] gets the character at index [pos] within string [s]. If
+    [pos] is negative or greater than the length of [s], returns the empty
+    string. If the string contains characters outside the range
+    [\u0000-\uffff], it will return the first 16-bit value at that position in
+    the string.
 
 {[
-  charAt 0, "Reason" = "R"
-  charAt( 12, "Reason") = "";
-  charAt( 5, {js|Rẽasöń|js} = {js|ń|js}
+  charAt "Reason" ~pos:0 = "R"
+  charAt "Reason" ~pos:12 = "";
+  charAt {js|Rẽasöń|js} ~pos:5 = {js|ń|js}
 ]}
 *)
 
-external charCodeAt : int -> float = "charCodeAt"
-[@@mel.send.pipe: t]
-(** [charCodeAt n s] returns the character code at position [n] in string [s]; the result is in the range 0-65535, unlke [codePointAt], so it will not work correctly for characters with code points greater than or equal to [0x10000].
-The return type is [float] because this function returns [NaN] if [n] is less than zero or greater than the length of the string.
+external charCodeAt : t -> pos:int -> float = "charCodeAt"
+[@@mel.send]
+(** [charCodeAt s ~pos] returns the character code at position [pos] in string
+    [s]; the result is in the range 0-65535, unlke [codePointAt], so it will
+    not work correctly for characters with code points greater than or equal to
+    [0x10000].
+    The return type is [float] because this function returns [NaN] if [pos] is
+    less than zero or greater than the length of the string.
 
 {[
-  charCodeAt 0 {js|😺|js} returns 0xd83d
-  codePointAt 0 {js|😺|js} returns Some 0x1f63a
+  charCodeAt {js|😺|js} ~pos:0 returns 0xd83d
+  codePointAt {js|😺|js} ~pos:0 returns Some 0x1f63a
 ]}
-
 *)
 
-(** [codePointAt n s] returns the code point at position [n] within string [s] as a [Some] value. The return value handles code points greater than or equal to [0x10000]. If there is no code point at the given position, the function returns [None].
+external codePointAt : t -> pos:int -> int option = "codePointAt"
```

Comment:
Maybe can rename `pos` to `index`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt#index

Reply:
I used `~pos` because that's what OCaml's `StdLabels` uses.